packages/fx-core/src/plugins/solution/fx-solution/v2/provisionLocal.ts

```diff
@@ -100,6 +107,16 @@ export async function provisionLocalResource(
     return new v2.FxPartialSuccess(localSettings, debugProvisionResult.error);
   }
 
+  if (isConfigUnifyEnabled()) {
+    const localEnvSetupResult = await setupLocalEnvironment(ctx, inputs, envInfo!);
+
+    if (localEnvSetupResult.isErr()) {
+      return new v2.FxPartialSuccess(localSettings, localEnvSetupResult.error);
+    }
+
+    setDataForLocal(envInfo!, localSettings);
+  }
```

Comment:
Try moving the corresponding part for the case when the feature flag is disable to a `else` statement, this would make the code much more readable and easier to clean-up the legacy implementation in the near future. For example:

```typescript
 if (isConfigUnifyEnabled()) {
    const localEnvSetupResult = await setupLocalEnvironment(ctx, inputs, envInfo!);

    if (localEnvSetupResult.isErr()) {
      return new v2.FxPartialSuccess(localSettings, localEnvSetupResult.error);
    }

    setDataForLocal(envInfo!, localSettings);
  } else {
    const debugProvisionResult = await setupLocalDebugSettings(ctx, inputs, localSettings);

    if (debugProvisionResult.isErr()) {
      return new v2.FxPartialSuccess(localSettings, debugProvisionResult.error);
    }
  }
```